File: nodejs/src/utility/firebase/connection.js
const connection = require("firebase-admin");
const serviceAccount = require("./credential.json");

connection.initializeApp({
  credential: connection.credential.cert(serviceAccount),
  databaseURL: "https://sample-project-267411.firebaseio.com"
});

const firebaseDB = connection.database()

module.exports = {
	connection,
	firebaseDB
}

File: golang/src/utility/firebase/connection.go
package config
 
import (
	"context"
	"log"
	firebase "firebase.google.com/go"
	"google.golang.org/api/option"
)
 
func Connection() (context.Context, *db.Client) {
	ctx := context.Background()
	conf := &firebase.Config{
        DatabaseURL: "https://sample-project-267411.firebaseio.com",
	}
	opt := option.WithCredentialsFile("utility/firebase/credential.json")
	app, err := firebase.NewApp(ctx, conf, opt)
	if err != nil {
        log.Fatalln("Error initializing app:", err)
	}

	client, err := app.Database(ctx)
	if err != nil {
        log.Fatalln("Error initializing database client:", err)
	}
	return ctx, client
}